src/telliot_core/contract/contract.py

```diff
@@ -75,9 +75,11 @@ async def read(
     async def write(
```

Comment:
It will be a challenge for someone to read through the code and understand the gas behavior. This function needs strong docstring paragraph(s) explaining the intent and how to use all of the gas price options, i.e.:
- How to choose between standard behavior (use EIP-1559) and legacy behavior
- default behavior (no arguments)
- Individual description of each gas argument